src/overlays/actors/ovl_En_Takaraya/z_en_takaraya.c

```diff
@@ -28,61 +47,360 @@ ActorInit En_Takaraya_InitVars = {
+            this->actor.textId = 0x77A;
+            gSaveContext.timerStates[TIMER_ID_MINIGAME_2] = TIMER_STATE_STOP;
+            func_80ADF6DC(this);
+        } else {
+            this->actor.textId = 0x77C;
+            if (CHECK_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED)) {
+                CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_WAIT);
+                CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED);
+                func_80ADEDF8(this);
+            } else {
+                func_80ADF6DC(this);
+            }
+        }
+    } else {
+        func_80ADEDF8(this);
+    }
+}
+
+void EnTakaraya_Destroy(Actor* thisx, PlayState* play) {
+    EnTakaraya* this = THIS;
+
+    Flags_UnsetSwitch(play, 5);
+    if (!this->unk2AD) {
+        CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_WAIT);
+        gSaveContext.timerStates[TIMER_ID_MINIGAME_2] = TIMER_STATE_STOP;
+    }
+}
+
+void func_80ADED8C(EnTakaraya* this) {
+    if (this->eyeTexIndex != 0) {
+        this->eyeTexIndex++;
+        if (this->eyeTexIndex == 4) {
+            this->eyeTexIndex = 0;
+        }
+    } else if (Rand_ZeroOne() < 0.02f) {
+        this->eyeTexIndex++;
+    }
+}
+
+void func_80ADEDF8(EnTakaraya* this) {
+    if (this->skelAnime.animation == &object_bg_Anim_001384) {
+        Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_00AD98, 5.0f);
+    }
+    this->actionFunc = func_80ADEE4C;
+}
+
+void func_80ADEE4C(EnTakaraya* this, PlayState* play) {
+    u16 textId;
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if ((&object_bg_Anim_00A280) == (*this).skelAnime.animation) {
+            Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_00AD98, 5.0f);
+        } else {
+            Animation_MorphToLoop(&this->skelAnime, &object_bg_Anim_009890, -4.0f);
+        }
+    }
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        if (Text_GetFaceReaction(play, FACE_REACTION_SET_TREASURE_CHEST_SHOP_GAL) == 0) {
+            Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_00A280, -4.0f);
+        }
+        func_80ADF03C(this);
+    } else if (Actor_IsFacingPlayer(&this->actor, 0x2000)) {
+        textId = Text_GetFaceReaction(play, FACE_REACTION_SET_TREASURE_CHEST_SHOP_GAL);
+        this->actor.textId = textId;
+        if (!(textId & 0xFFFF)) {
+            this->actor.textId = D_80ADFB2C[(void)0, gSaveContext.save.playerForm];
+        }
+        this->unk2B0 = gSaveContext.save.playerForm + this->switchFlag;
+        Actor_OfferTalk(&this->actor, play, 120.0f);
+    }
+}
+
+void func_80ADEF74(EnTakaraya* this, PlayState* play) {
+    u8 var_v1;
+
+    if (Flags_GetSwitch(play, this->unk2B0)) {
+        var_v1 = D_80ADFB38[gSaveContext.save.playerForm][1];
+    } else {
+        var_v1 = D_80ADFB38[gSaveContext.save.playerForm][0];
+    }
+    if (gSaveContext.save.playerForm) {}
+    Actor_SpawnAsChildAndCutscene(&play->actorCtx, play, ACTOR_OBJ_TAKARAYA_WALL, 0.0f, 0.0f, 0.0f, 0, 0, 5,
```

Comment:
```suggestion
    }
    
    //! FAKE:
    if (gSaveContext.save.playerForm) {}

    Actor_SpawnAsChildAndCutscene(&play->actorCtx, play, ACTOR_OBJ_TAKARAYA_WALL, 0.0f, 0.0f, 0.0f, 0, 0, 5,
```
Could you post a decompme of this function? I suspect you need `D_80ADFB38[((void)0, gSaveContext.save.playerForm)]` above which can use the `GET_PLAYER_FORM` macro.

Also when there's a fake match, you want the `//! FAKE:` label and to isolate the fake with spacing

Reply:
Yeah, exactly. You should be using the macros for the data access like this. That will get rid of the fake match!
https://decomp.me/scratch/4y7uT